types/tools/tool-settings.d.ts

```diff
@@ -19,7 +19,7 @@ export interface ToolboxConfig {
 /**
  * Object passed to the Tool's constructor by {@link EditorConfig#tools}
  */
-export interface ToolSettings {
+export interface ToolSettings <T extends object = any> {
```

Comment:
`T` should be named as `Config`. And this type variable should be documented via `@template` tag ([example](https://github.com/codex-team/ctproto/blob/main/types/message.ts))